pkg/frontend/middleware/msi.go

```diff
@@ -0,0 +1,24 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/Azure/msi-dataplane/pkg/dataplane"
+)
+
+const (
+	mockIdentityURL    = "https://bogus.identity.azure-int.net/subscriptions/00000000-0000-0000-0000-000000000000/resourcegroups/rg/providers/Microsoft.ApiManagement/service/test/credentials?tid=00000000-0000-0000-0000-000000000000&oid=00000000-0000-0000-0000-000000000000&aid=00000000-0000-0000-0000-000000000000"
+	mockTenantIDEnvVar = "MOCK_MSI_TENANT_ID"
```

Comment:
I view the middleware as passive, meaning it's just piping an environment variable.

This allows us the following:
- In our E2E testing, we can add a test where MSI tenant ID header is empty, so the rest of our code handles that properly.
- We already have code to validate tenant ID: https://github.com/Azure/ARO-RP/blob/13e4360be2dbbe8329e45c8ae7946b25a345392d/pkg/frontend/openshiftcluster_putorpatch.go#L148

Reply:
I didn't think about the fact that E2E will use a mock MSI. That part makes sense, and so does your point about the tenant ID validation. But where are you thinking that we'll actually read the value of that env var? I still don't get that part.

Reply:
the value is read directly by the middleware. Env var is set via the scripts I worked on previously.

Reply:
I guess I'm having trouble understanding then, because my reading of the code is that the middleware sets the tenant ID to the literal string `"MOCK_MSI_TENANT_ID"` and the env var is never read. Could you ELI5 and point me to exactly where it's being read?